### PR TITLE
Implement FGetObject and FPutObject

### LINKED ISSUE #267: Implement fputobject fgetobject

### PR DESCRIPTION
**Pull request by harshavardhana**
Fixes #267

### CODE REVIEW

**Comment by harshavardhana** [comment]
FPutObject is not implemented yet, do not merge.  It will be subsequently added right here.

**Comment by harshavardhana** [comment]
FPutObject is also implemented, but requires a lot of testing, i will confirm once it starts working properly.

**Comment by harshavardhana** [comment]
FPutObject and FGetObject are implemented and working - please review.

**Comment by harshavardhana** [comment]
> should we auto detect contentType as we have the file extension type?

No it is not that straightforward, it is better for user to set.

File: api-fput-object.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+		completedPart.ETag = objPart.ETag
+		completeMultipartUpload.Parts = append(completeMultipartUpload.Parts, completedPart)
+		// Save total uploaded size which will be incremented later.
+		totalUploadedSize += objPart.Size
+		// Increment additively to verify holes in next iteration.
+		partNumber++
+	}
+
+	// Return.
+	return partNumber, totalUploadedSize, completeMultipartUpload, nil
```
Comment (abperiasamy) [inline api-fput-object.go]: partNumber should be partsCount

File: api-fput-object.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+		// Save the new upload id.
+		uploadID = initMultipartUploadResult.UploadID
+	}
+	return uploadID, nil
+}
+
+// getMultipartStat gather next part number, total uploaded size and list of parts.
+// TODO: reduce number of results.
+func (c Client) getMultipartStat(bucketName, objectName, uploadID string) (int, int64, completeMultipartUpload, error) {
+	// Input validation.
```
Comment (abperiasamy) [inline api-fput-object.go]: returning 4 variables is fine, but in cases like these int and int64 does not convey what it stands for. Either we can document them or declare variables here (preferred approach).

File: api-fget-object.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	}
+
+	// Seek to current position for incoming reader.
+	objectReader, objectStat, err := c.getObject(bucketName, objectName, st.Size(), 0)
+	if err != nil {
+		return err
+	}
+
+	// Write to the part file.
+	if _, err = io.CopyN(filePart, objectReader, objectStat.Size); err != nil {
```
Comment (abperiasamy) [inline api-fget-object.go]: must verify if written != st.Size and return errShortRead

**Pull request merged**